### ISSUE TITLE
RuntimeError when concatenating frames wtih IndexHierarchy and Year dtype

### ISSUE THREAD

**Issue by ForeverWintr** [InvestmentSystems/static-frame#512]
**Description**

Two issues: 

1. I don't think this should fail.
2. If it does fail, I think it should raise `ErrorInitIndex` or a `ValueError` subclass.

**Example**

Provide an interactive Python session, or Python code, to demonstrate the issue. Please keep example data to a minimum.

```python
import frame_fixtures as ff
import static_frame as sf

f = ff.parse("f(Fg)|v(int,bool,str)|i((IY,ID),(dtY,dtD))|c(ISg,dts)|s(4,2)")

f1 = f.iloc[:2]
f2 = f.iloc[2:]

sf.Frame.from_concat((f1, f2)) # RuntimeError

```

Traceback:

```
  File "/home/rutherford/.env38/lib/python3.8/site-packages/static_frame/core/frame.py", line 521, in from_concat
    index = index_many_concat(
  File "/home/rutherford/.env38/lib/python3.8/site-packages/static_frame/core/container_util.py", line 1268, in index_many_concat
    return _index_many_to_one(indices,
  File "/home/rutherford/.env38/lib/python3.8/site-packages/static_frame/core/container_util.py", line 1257, in _index_many_to_one
    return constructor(array, #type: ignore
  File "/home/rutherford/.env38/lib/python3.8/site-packages/static_frame/core/index_hierarchy.py", line 584, in from_labels
    indices = [constructor(hash_map)
  File "/home/rutherford/.env38/lib/python3.8/site-packages/static_frame/core/index_hierarchy.py", line 584, in <listcomp>
    indices = [constructor(hash_map)
  File "/home/rutherford/.env38/lib/python3.8/site-packages/static_frame/core/container_util.py", line 413, in func
    return index_from_optional_constructor(value,
  File "/home/rutherford/.env38/lib/python3.8/site-packages/static_frame/core/container_util.py", line 383, in index_from_optional_constructor
    return explicit_constructor(value) #type: ignore
  File "/home/rutherford/.env38/lib/python3.8/site-packages/static_frame/core/index_datetime.py", line 70, in __init__
    Index.__init__(self,
  File "/home/rutherford/.env38/lib/python3.8/site-packages/static_frame/core/index.py", line 330, in __init__
    self._map = FrozenAutoMap(labels_for_automap) if self.STATIC else AutoMap(labels_for_automap)
  File "/home/rutherford/.env38/lib/python3.8/site-packages/static_frame/core/index.py", line 307, in <genexpr>
    labels = (to_datetime64(v, dtype_extract) for v in labels)
  File "/home/rutherford/.env38/lib/python3.8/site-packages/static_frame/core/util.py", line 1702, in to_datetime64
    raise RuntimeError(f'value ({dt}) is not a supported dtype ({dtype})')
builtins.RuntimeError: value (36685-01-01) is not a supported dtype (datetime64[Y])
```

**Platform**

Run the following function (static-frame >= 0.8.1) and provide the results to define your platform and environment:

```python
>>> import static_frame as sf
>>> sf.Platform.display()
<Series: platform>
<Index>
platform           Linux-5.13.0-52-generic-x86_64-with
sys.version        3.8.12 (default, Jul  5 2022, 10:53:07) [GCC 7.5.0]
static-frame       0.9.6
numpy              1.17.4
pandas             0.24.2
xlsxwriter         1.1.5
openpyxl           3.0.9
xarray             0.18.0
tables             3.6.1
pyarrow            0.17.0
msgpack            (1, 0, 0)
msgpack_numpy      None
<<U13>             <object>
```

**Comment by flexatone** [comment]
@ForeverWintr , agreed this should not fail. @chaburkland , thanks for the triage, I will look into it.

**Comment by flexatone** [comment]
This issue is corrected and will be part of the 0.9.8 release.
